UnityProject/Assets/Scripts/Managers/PlayerList.cs

```diff
@@ -302,7 +312,47 @@ public List<ConnectedPlayer> GetAllByUserID(string byUserID, bool includeOffline
  		return newone.FindAll(player => player.UserId == byUserID);
 	}
 
-	private ConnectedPlayer getInternal(Func<ConnectedPlayer, bool> condition)
+	/// <summary>
+	/// Check logged in and logged off players
+	/// </summary>
+	/// <param name="condition"></param>
+	/// <returns></returns>
+	private ConnectedPlayer GetInternalAll(Func<ConnectedPlayer, bool> condition)
+	{
+		var connectedPlayer = GetInternalLoggedIn(condition);
+
+		if(connectedPlayer.Equals(ConnectedPlayer.Invalid))
+		{
+			connectedPlayer = GetInternalLoggedOff(condition);
+		}
+
+		return connectedPlayer;
+	}
+
+	/// <summary>
+	/// Check logged in players
+	/// </summary>
+	/// <param name="condition"></param>
+	/// <returns></returns>
+	private ConnectedPlayer GetInternalLoggedIn(Func<ConnectedPlayer, bool> condition)
+	{
+		for (var i = 0; i < loggedIn.Count; i++)
+		{
+			if (condition(loggedIn[i]))
+			{
+				return loggedIn[i];
+			}
+		}
+
+		return ConnectedPlayer.Invalid;
+	}
+
+	/// <summary>
+	/// Check logged off players
+	/// </summary>
+	/// <param name="condition"></param>
+	/// <returns></returns>
+	private ConnectedPlayer GetInternalLoggedOff(Func<ConnectedPlayer, bool> condition)
```

Comment:
This is identical to `GetInternalLoggedIn()`?

Reply:
dont worry im an idiot